internal/controller/pgupgrade/jobs.go

```diff
@@ -81,49 +81,52 @@ func upgradeCommand(spec *v1beta1.PGUpgradeSettings, fetchKeyCommand string) []s
 		// one major version to another. Additional information concerning the
 		// steps used and command flag specifics can be found in the documentation:
 		// - https://www.postgresql.org/docs/current/pgupgrade.html
 
-		// To begin, we first move to the mounted /pgdata directory and create a
-		// new version directory which is then initialized with the initdb command.
-		`cd /pgdata || exit`,
-		`echo -e "Step 1: Making new pgdata directory...\n"`,
-		`mkdir /pgdata/pg"${new_version}"`,
-		`echo -e "Step 2: Initializing new pgdata directory...\n"`,
-		initdb,
-
-		// Before running the upgrade check, which ensures the clusters are compatible,
-		// proper permissions have to be set on the old pgdata directory and the
-		// preload library settings must be copied over.
-		`echo -e "\nStep 3: Setting the expected permissions on the old pgdata directory...\n"`,
-		`chmod 750 /pgdata/pg"${old_version}"`,
-		`echo -e "Step 4: Copying shared_preload_libraries setting to new postgresql.conf file...\n"`,
-		`echo "shared_preload_libraries = '$(/usr/pgsql-"""${old_version}"""/bin/postgres -D \`,
-		`/pgdata/pg"""${old_version}""" -C shared_preload_libraries)'" >> /pgdata/pg"${new_version}"/postgresql.conf`,
-
-		// Before the actual upgrade is run, we will run the upgrade --check to
-		// verify everything before actually changing any data.
-		`echo -e "Step 5: Running pg_upgrade check...\n"`,
-		`time /usr/pgsql-"${new_version}"/bin/pg_upgrade --old-bindir /usr/pgsql-"${old_version}"/bin \`,
-		`--new-bindir /usr/pgsql-"${new_version}"/bin --old-datadir /pgdata/pg"${old_version}"\`,
-		` --new-datadir /pgdata/pg"${new_version}" --check` + argMethod + argJobs,
-
-		// Assuming the check completes successfully, the pg_upgrade command will
-		// be run that actually prepares the upgraded pgdata directory.
-		`echo -e "\nStep 6: Running pg_upgrade...\n"`,
-		`time /usr/pgsql-"${new_version}"/bin/pg_upgrade --old-bindir /usr/pgsql-"${old_version}"/bin \`,
-		`--new-bindir /usr/pgsql-"${new_version}"/bin --old-datadir /pgdata/pg"${old_version}" \`,
-		`--new-datadir /pgdata/pg"${new_version}"` + argMethod + argJobs,
-
-		// Since we have cleared the Patroni cluster step by removing the EndPoints, we copy patroni.dynamic.json
-		// from the old data dir to help retain PostgreSQL parameters you had set before.
-		// - https://patroni.readthedocs.io/en/latest/existing_data.html#major-upgrade-of-postgresql-version
-		`echo -e "\nStep 7: Copying patroni.dynamic.json...\n"`,
-		`cp /pgdata/pg"${old_version}"/patroni.dynamic.json /pgdata/pg"${new_version}"`,
-
-		`echo -e "\npg_upgrade Job Complete!"`,
+		// pg_upgrade writes its files in "${new_data}/pg_upgrade_output.d" since PostgreSQL v15.
+		// Change to a writable working directory to be compatible with PostgreSQL v14 and earlier.
+		//
+		// https://www.postgresql.org/docs/release/15#id-1.11.6.20.5.11.3
+		`cd "${data_volume}" || exit`,
+
+		`section 'Initializing new data directory...'`,
```

Comment:
Can we keep the step numbers or maybe even better, say n/N steps to give a sense of where in the process we are.